src/components/contest/ContestPage.tsx

```diff
@@ -287,18 +289,20 @@ const ContestPage = () => {
           </div>
         </div>
       </div>
-      <footer className={"pt-2 " + state.appState.theme.bg}>
-        <Pagination
-          pageSelected={(e) => setSelected(e)}
-          perPage={filter.perPage}
-          selected={selected}
-          theme={state.appState.theme}
-          totalCount={contestList.contests.length}
-          pageSize={(e) => {
-            setFilter({ ...filter, perPage: e });
-          }}
-        />
-      </footer>
+      {!isRandomClicked && (
```

Comment:
```suggestion
      {randomContest === -1 && (
```
If randomContest is -1, then a random contest is not selected. Otherwise, it will store the index of a contest. You can remove the isRandomClicked variable. Also, currently, when the user cancels, random pagination is not shown again in your implementation. If you implement my suggestion, that bug will be fixed.

Reply:
Yes, using randomContest === -1 is better than using another state variable, as this randomContest === -1, will maintain the proper check